lib/libnfs.c

```diff
@@ -2238,3 +2383,59 @@ rpc_null_task(struct rpc_context *rpc, int program, int version, rpc_cb cb,
 
 	return pdu;
 }
+
+#ifdef HAVE_TLS
+/*
+ * Call this in place of rpc_null_task() if user wants TLS security for the RPC
+ * connection. Since we only ever send AUTH_TLS NULL RPC for NFS_PROGRAM it does
+ * not need the program parameter.
+ * Callback 'cb' is called not after we get the reply for this AUTH_TLS NULL RPC,
+ * but after the TLS handshake completes (success or failure).
+ */
+struct rpc_pdu *
+rpc_null_task_authtls(struct rpc_context *rpc, int nfs_version, rpc_cb cb,
+		      void *private_data)
+{
+	struct rpc_pdu *pdu;
+	struct tls_cb_data *data;
+
+	/* Must be called only for secure connections to NFS_PROGRAM version 3 or 4 */
+	assert(rpc->use_tls);
+	assert(nfs_version == NFS_V3 || nfs_version == NFS_V4);
+
+	data = calloc(1, sizeof(*data));
+	if (data == NULL) {
+		rpc_set_error(rpc, "Out of memory. Failed to allocate tls_cb_data "
+			      "for AUTH_TLS NULL call");
+		return NULL;
+	}
+	data->cb 	   = cb;
+	data->private_data = private_data;
+
+	/*
+	 * Set MSbit in procedure number to convey use of AUTH_TLS
+	 * This should not interfere with valid procedure numbers as they are all
+	 * small numbers.
+	 */
+	pdu = rpc_allocate_pdu(rpc, NFS_PROGRAM, nfs_version, (0 | (1U << 31)),
```

Comment:
In rpc_allocate_pdu2() we use 0x80000000U to describe this.  Can you change (1 <<31)  to that so we have the same constant/expression in both places?